File: maven-leetcode/src/main/java/leetcode/Num160.java
package leetcode;

public class Num160 {
    /**思路：可先分别统计两个链表长度，然后再让长的链表指针先走几步，再两个同时走，第一次相遇的位置即为交叉的位置
     * 否则不交叉*/
    public class Solution {
        public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
            if(headA == null || headB == null){return null;}
            int aLen = 0;
            int bLen = 0;
            ListNode aPoint = headA;
            ListNode bPoint = headB;
            while(aPoint != null){
                aLen++;
                aPoint = aPoint.next;
            }
            while(bPoint != null){
                bLen++;
                bPoint = bPoint.next;
            }
            if(aLen < bLen){
                for(int i = 0; i < bLen - aLen; i++){
                    headB = headB.next;
                }
            } else if(aLen > bLen){
                for(int i = 0; i < aLen - bLen; i++){
                    headA = headA.next;
                }
            }
            while(headA != null && headB != null){
                if(headA == headB){
                    return headA;
                }
                headA = headA.next;
                headB = headB.next;
            }
            return null;

        }
    }
}

File: learnPython/num160.py
class Solution(object):
    def getIntersectionNode(self, headA, headB):
        """
        :type head1, head1: ListNode
        :rtype: ListNode
        """
        lenA = 0
        lenB = 0
        pA = headA
        pB = headB
        while pA != None:
            lenA = lenA + 1
            pA = pA.next
        while pB != None:
            lenB = lenB + 1
            pB = pB.next
        if lenA == 0 or lenB == 0:
            return None
        if lenA > lenB:
            for i in range(0,lenA-lenB):
                headA = headA.next
        elif lenA < lenB:
            for i in range(0,lenB - lenA):
                headB = headB.next
        while headA!=None:
            if headA == headB:
                return headA
            headA = headA.next
            headB = headB.next
        return None